dev/NumberBox/NumberBox.cpp

```diff
@@ -45,6 +45,20 @@ NumberBox::NumberBox()
     LostFocus({ this, &NumberBox::OnNumberBoxLostFocus });
 
     SetDefaultStyleKey(this);
+    SetDefaultInputScope();
+}
+
+void NumberBox::SetDefaultInputScope()
+{
+    // Sets the default value of the InputScope property.
+    // Note that InputScope is a class that cannot be set to a default value within the IDL.
+    winrt::InputScopeName inputScopeName = winrt::InputScopeName(winrt::InputScopeNameValue::Number);
+    winrt::InputScope inputScope = winrt::InputScope();
+    inputScope.Names().Append(inputScopeName);
```

Comment:
>Names [](start = 15, length = 5)

I'm not that familiar with inputScope, does this mean that you can apply multiple input scopes? like you can have number and currency at the same time.

Reply:
> does this mean that you can apply multiple input scopes?
Yes, you can apply multiple name values to the same input scope.

It's certainly an interesting design decision, one which I've never understood myself. However, my assumption has always been it's for the cases where the user can switch the keyboard. It simply limits what keyboards you are allowed to switch between. You might have to reach out to the Windows team to understand the thinking here though. I don't recall ever seeing documented reasons why this is a collection.

Reply:
@MikeHillberg do you happen to know?